<launch>
	<arg name="sub_msg_to_robot_topic_name"       default="/interactive_cleanup/message/to_robot" />
	<arg name="pub_msg_to_moderator_topic_name"   default="/interactive_cleanup/message/to_moderator" />
	<arg name="sub_joint_state_topic_name"        default="/hsrb/joint_states" />
	<arg name="pub_base_twist_topic_name"         default="/hsrb/opt_command_velocity" />
	<arg name="pub_arm_trajectory_topic_name"     default="/hsrb/arm_trajectory_controller/command" />
	<arg name="pub_gripper_trajectory_topic_name" default="/hsrb/gripper_trajectory_controller/command" />
	<arg name="sub_laser_scan_topic_name"         default="/hsrb/base_scan" />
	<arg name="rgbd_camera"                       default="head_rgbd_sensor" />
	<arg name="sigverse_ros_bridge_port"          default="50001" />
	<arg name="ros_bridge_port"                   default="9090" />
	<arg name="map_file"                          default="/home/kamerider/sim_ws/src/CleanUp/interactive_cleanup/maps/interactive_clean_up_test.yaml" />
	<!-- yaml中写的路径是绝对路径，移植时需要修改 -->

	<!-- 加载按键操纵节点 -->
	<node name="teleop_key_interactive_cleanup" pkg="interactive_cleanup" type="teleop_key_interactive_cleanup" output="screen" launch-prefix="xterm -e">
		<param name="sub_msg_to_robot_topic_name"        type="string" value="$(arg sub_msg_to_robot_topic_name)" />
		<param name="pub_msg_to_moderator_topic_name"    type="string" value="$(arg pub_msg_to_moderator_topic_name)" />
		<param name="sub_joint_state_topic_name"         type="string" value="$(arg sub_joint_state_topic_name)" />
		<param name="pub_base_twist_topic_name"          type="string" value="$(arg pub_base_twist_topic_name)" />
		<param name="pub_arm_trajectory_topic_name"      type="string" value="$(arg pub_arm_trajectory_topic_name)" />
		<param name="pub_gripper_trajectory_topic_name"  type="string" value="$(arg pub_gripper_trajectory_topic_name)" />
	</node>

	<!-- 加载Unity-Ros连接桥 -->
	<include file="$(find rosbridge_server)/launch/rosbridge_websocket.launch" >
		<arg name="port" value="$(arg ros_bridge_port)"/>
	</include>
	<group ns="sigverse_ros_bridge">
		<node name="sigverse_ros_bridge" pkg="sigverse_ros_bridge" type="sigverse_ros_bridge" args="$(arg sigverse_ros_bridge_port)"/>
	</group>

	<!-- 加载RGBD传感器 -->
	<group ns="hsrb">
		<include file="$(find realsense_camera)/launch/r200_nodelet_rgbd.launch">
			<arg name="camera"      value="$(arg rgbd_camera)"/>
			<arg name="load_driver" value="false"/>
		</include>
	</group>

	<!-- 加载rviz默认界面配置 -->
	<node pkg="rviz" type="rviz" name="rviz" args="-d $(find interactive_cleanup)/launch/hsr.rviz"/>

	<!-- 加载地图 -->
  	<node pkg="map_server" type="map_server" name="map_server" args="$(arg map_file)" />


	<!-- 加载amcl-->
	<arg name="initial_pose_x" default="0.0"/>
	<arg name="initial_pose_y" default="0.0"/>
	<arg name="initial_pose_a" default="0.0"/>
	<include file="$(find turtlebot_navigation)/launch/includes/amcl/amcl.launch.xml">
		<arg name="initial_pose_x" value="$(arg initial_pose_x)"/>
		<arg name="initial_pose_y" value="$(arg initial_pose_y)"/>
		<arg name="initial_pose_a" value="$(arg initial_pose_a)"/>
	</include>

	<!-- 加载move_base -->
	<arg name="cmd_vel_topic" default="/hsrb/opt_command_velocity" />
	<arg name="odom_topic" default="odom" />
	<node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
		<param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS" />
		<rosparam file="$(find interactive_cleanup)/param/costmap_common_params_hsr.yaml" command="load" 	ns="global_costmap" />
		<rosparam file="$(find interactive_cleanup)/param/costmap_common_params_hsr.yaml" command="load" ns="local_costmap" />
		<rosparam file="$(find interactive_cleanup)/param/local_costmap_params.yaml" command="load" />
		<rosparam file="$(find interactive_cleanup)/param/global_costmap_params.yaml" command="load" />
		<rosparam file="$(find interactive_cleanup)/param/move_base_params.yaml" command="load" />
		<rosparam file="$(find interactive_cleanup)/param/dwa_local_planner_params.yaml" command="load" />
		<remap from="cmd_vel" to="$(arg cmd_vel_topic)"/>
		<remap from="odom" to="$(arg odom_topic)"/>
	</node>
</launch>
